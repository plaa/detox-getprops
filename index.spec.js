jest.mock('detox');

const { expect: expectDetox } = require('detox');
const { getProps, parseMessage } = require('./index');
const { iosMessages, androidMessages } = require('./exampleMessages');

describe('getProps', () => {
  it('should return iOS props', async () => {
    const elem = 'boom';
    expectDetox.mockImplementation(() => {
      throw new Error(iosMessages['Step, One']);
    });
    const props = await getProps(elem);
    expect(props).toMatchSnapshot();
  });

  it('should return Android props', async () => {
    const elem = 'boom';
    expectDetox.mockImplementation(() => {
      throw new Error(androidMessages['Step, One']);
    });
    const props = await getProps(elem);
    expect(props).toMatchSnapshot();
  });

  it('should throw exception for unknown exception messages', async () => {
    const elem = 'boom';
    expectDetox.mockImplementation(() => {
      throw new Error('Something');
    });
    let threw = false;
    try {
      await getProps(elem);
    } catch (e) {
      threw = true;
    }
    expect(threw).toBeTruthy();
  });
});

describe('parseMessage', () => {
  it('should parse iOS matching string', () => {
    Object.entries(iosMessages).forEach(([text, input]) => {
      const output = parseMessage(input);
      expect(output.text).toEqual(text);
      expect(output).toMatchSnapshot();
    });
  });

  it('should parse Android matching string', () => {
    Object.entries(androidMessages).forEach(([text, input]) => {
      const output = parseMessage(input);
      expect(output.text).toEqual(text);
      expect(output).toMatchSnapshot();
    });
  });

  it('should throw error on unknown strings', () => {
    expect(() => parseMessage('foo')).toThrow();
  });
});
