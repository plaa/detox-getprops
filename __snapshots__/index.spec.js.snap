// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseMessage should parse iOS matching string 1`] = `
Object {
  "AX": "Y",
  "AX.activationPoint": "{187.5, 298.25}",
  "AX.focused": "N",
  "AX.frame": "{{24, 283.5}, {327, 29.5}}",
  "AX.id": "heading",
  "AX.label": "Step' One",
  "AX.traits": "UIAccessibilityTraitStaticText",
  "frame": "{{24, 0}, {327, 29.5}}",
  "text": "Step' One",
  "type": "RCTTextView",
}
`;

exports[`parseMessage should parse iOS matching string 2`] = `
Object {
  "AX": "Y",
  "AX.activationPoint": "{187.5, 298.25}",
  "AX.focused": "N",
  "AX.frame": "{{24, 283.5}, {327, 29.5}}",
  "AX.id": "heading",
  "AX.label": "Step, One",
  "AX.traits": "UIAccessibilityTraitStaticText",
  "frame": "{{24, 0}, {327, 29.5}}",
  "text": "Step, One",
  "type": "RCTTextView",
}
`;
